src/convertToObject.js

```diff
@@ -6,7 +6,24 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const str = sourceString
+    .replace(/^\s*[\r\n]/gm, '')
+    .replace(/\s*;\s*/g, ';')
+    .replace(/\s*:\s*/g, ': ')
+    .trim();
+
+  const cleanArray = str.split(';').filter((elem) => elem.length > 0);
+  const obj = {};
+
+  for (let i = 0; i < cleanArray.length; i++) {
+    const properties = cleanArray[i].split(':');
+    const key = properties[0].trim();
+    const value = properties[1].trim();
```

Comment:
There is a potential issue here if the `properties` array does not have exactly two elements. This can happen if the input string is malformed or if there are missing colons. Consider adding a check to ensure `properties` has the expected length before accessing its elements.